core/flyout_base.js

```diff
@@ -612,11 +612,17 @@ Blockly.Flyout.prototype.createBlock_ = function(blockXml) {
  * @private
  */
 Blockly.Flyout.prototype.getBlockXml_ = function(blockInfo) {
-  var blockXml = blockInfo['blockxml'];
-  if (blockXml) {
+  var blockXml = null;
+  // All blockInfo will have type, so check for blockxml first.
+  if (blockInfo['blockxml']) {
     blockXml = Blockly.Xml.textToDom(blockInfo['blockxml']);
+  } else if (blockInfo['type']) {
+    var blockXml = Blockly.utils.xml.createElement('xml');
+    blockXml.setAttribute('type', blockInfo['type']);
+    blockXml.setAttribute('gap', blockInfo['gap']);
```

Comment:
`gap` and `disabled` are also not on the blockInfo type ``Blockly.utils.toolbox.Block``.